sources/api/controllers/CalendarController.ts

```diff
@@ -97,17 +99,34 @@ export async function deleteCalendar(req: Request, res: Response) {
 }
 
 export async function createEvent(req: Request, res: Response) {
+    const requestingUser = getRequestingUser(req);
+    const name = req.body.name;
+    const description = req.body.description;
+    const location = req.body.location;
+    const start_time = req.body.start_time;
+    const end_time = req.body.end_time;
+    const calendar_id = req.params.calendar_id;
+    if (!name || !description || !location || !start_time || !end_time) {
+        throw new ApiException(400, "Bad request");
+    }
+
+    const conn = await DbConnection.getConnection();
+    let calendar = await Calendar.getCalendarWithOwners(conn, calendar_id);
+
+    if (!calendar || !calendar.isCalendarOwner(requestingUser)) {
+        throw new ApiException(404, "Calendar not found");
+    }
+
+    let event = new Event(name, description, location, calendar, start_time, end_time);
+    if (!event.isValid()) {
+        throw new ApiException(400, "Invalid fields in Event");
+    }
+
```

Comment:
I'd say "no new line here" 😃

Reply:
You don't make sense here, why no new line here and you want one after conn initialization

Reply:
I want to have logic blocs in the function

Here:
- Get the requesting user
- Get request params and check integrity
- Get connection
- Make sure the requesting user can access the calendar
- created and save the event
- response

Reply:
okok i see